packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.tsx

```diff
@@ -299,9 +258,14 @@ useTreeViewItems.getDefaultizedParams = ({ params }) => ({
   itemChildrenIndentation: params.itemChildrenIndentation ?? '12px',
 });
 
-useTreeViewItems.wrapRoot = ({ children, instance }) => {
+useTreeViewItems.wrapRoot = ({ children, store }) => {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const contextValue = React.useCallback(
```

Comment:
There is a hidden usage of a selector in render without `useSelector` here... 
I'll think about a way to fix it before merging